source/develop/release-management/features/gluster-self-heal-monitoring.html.md

```diff
@@ -41,21 +41,51 @@ Host Maintenance and Fencing will be enabled/disabled based on the following cri
 | No Unsynced Entries | Quorum available without this Node | Allowed |
 | No Unsynced Entries | Quorum not available without this Node | Not Allowed |
 
+ More info about standard host fencing is available at http://old.ovirt.org/Automatic_Fencing#Automatic_Fencing http://old.ovirt.org/Fence_kdump https://www.youtube.com/watch?v=V1JQtmdleaM
+
 ## Entity Changes
 Following entities will be changed as part of Gluster self heal monitoring.
 
 ### Gluster_volume_bricks
   Following columns will be added to gluster_volume_bricks table.
   unsynced_entries  - integer - No.of unsynced entries in the brick.
   unsynced_entires_history - text - History of unsynced entries in the brick. It will a list of comma separated values.
-  
-### Gluster_volumes_view
-‘heal_status’ column will be be added to gluster_volumes_view. It will be computed based on unsynced_entries in the brick. If there is any unsynced_entry in any of the bricks in the volume then ‘heal_status’ will be ‘Needs_Healing’.
 
+## Change in BLL Commands
+###VdsNotRespondingTreatmentCommand (for Host fencing)
+   VdsNotRespondingTreatmentCommand is the central BLL command which takes care of varios host fencing related activites. This will be changed to consider gluster self-heal information during host fencing. shouldFencingBeSkipped method will be changed and following logic will be added.
+   
+   if (host supports both gluster & virt services)
+      if(all bricks are down)
+        allow host fencing
+      else(some || all bricks are up)
+        if (unsync'd entries  > 0 for any up brick) //Entris needs to be healed from this host
+          skip host fencing
+        else
+          if (quorum is maintained (bricks and nodes for ALL volumes) with this host down)
+            allow host fencing
+          else
+            skip host fencing
+
```

Comment:
Sure, here is the whole flow:

1. Please take a look at org.ovirt.engine.core.common.businessentities.FencingPolicy, for sanlock lease there's skipFencingIfSDActive attribute (all FencingPolicy attributes are stored in clusters table)
2. Some attributes of FencingPolicy are checked in VdsNotRespondingTreatmentCommand.shouldFencingBeSkipped(), but skipFencingIfSDActive is passed down the stack
3. FencingPolicy instance is passed through NotRespondingTreatmentCommand -> RestartVdsCommand -> FenceActionExecutor -> FenceAgentExecutor where FencingPolicy is converted in convertFencingPolicy() into a Map which is passed to fenceNode VDSM verb
4. On VDSM please take a look at fenceNode in API.py, where in should_fence method we check sanlock lease status